chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/complaints/ComplaintsReportCsvCreator.java

```diff
@@ -0,0 +1,270 @@
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.lang3.BooleanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.domain.complaint.ComplaintListingMap;
+import gov.healthit.chpl.domain.surveillance.Surveillance;
+import gov.healthit.chpl.util.DateUtil;
+import gov.healthit.chpl.util.Util;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2(topic = "complaintsReportJobLogger")
+@Component
+public class ComplaintsReportCsvCreator {
+
+    private Environment env;
+
+    @Autowired
+    public ComplaintsReportCsvCreator(Environment env) {
+        this.env = env;
+    }
+
+    private static final String NEW_LINE_SEPARATOR = "\n";
+
+    public File createCsvFile(List<ComplaintsReportItem> reports) throws IOException {
+        CSVFormat csvFileFormat = CSVFormat.DEFAULT.builder()
+                .setRecordSeparator(NEW_LINE_SEPARATOR)
+                .build();
+
+        File csvFile = getOutputFile();
+        try (FileWriter fileWriter = new FileWriter(csvFile);
+                CSVPrinter csvFilePrinter = new CSVPrinter(fileWriter, csvFileFormat)) {
+
+            csvFilePrinter.printRecord(getHeaderRow());
+
+            reports.stream()
+                .sorted((cri1, cri2) -> cri1.getComplaint().getId().compareTo(cri2.getComplaint().getId()))
+                .forEach(report -> printRows(csvFilePrinter, report));
+        }
+        return csvFile;
+    }
+
+    private File getOutputFile() {
+        File temp = null;
+        try {
+            temp = File.createTempFile(getFilename(), ".csv");
+            temp.deleteOnExit();
+        } catch (IOException ex) {
+            LOGGER.error("Could not create temporary file " + ex.getMessage(), ex);
+        }
+
+        return temp;
+    }
+
+    private List<String> getHeaderRow() {
+        return Arrays.asList(
+                "Complaint ID",
+                "ONC-ACB Complainant Type Name",
+                "Complainant Type - Other",
+                "ONC Complaint ID",
+                "ONC-ACB Complaint ID",
+                "Received Date",
+                "Complaint Summary",
+                "Actions/Responses",
+                "Complainant Contacted",
+                "Developer Contacted",
+                "ONC-ATL Contacted",
+                "Informed ONC Per §170.523(S)",
+                "Closed Date",
+                "Developer Name",
+                "Product Name",
+                "Version",
+                "Associated Listings",
+                "Associated Criteria",
+                "Associated Surveillance",
+                "Surveillance Result",
+                "Non-Conformity Yype",
```

Comment:
```suggestion
                "Non-Conformity Type",
```

Will approve this PR, but this should change eventually